File: L/submissions/tle/tle.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solve(long long n) {
    assert(1ll <= n && n <= 1e18);
    vector<int> ans;

    for (long long x = 1; x * (x + 1) <= n; x++) {
        if (n% (x * x + x) == 0) {
            ans.push_back(x);
        }
    }

    return ans;
}

int main() {
    int t;
    long long n;

    cin >> t;
    assert(1 <= t && t <= 100);

    for (int cs = 1; cs <= t; cs++) {
        cin >> n;
        vector<int> ans = solve(n);
        sort(ans.begin(), ans.end());

        if (ans.size() == 0) {
            cout << "-1" << '\n';
        }
        for (size_t i = 0; i < ans.size(); i++) {
            cout << ans[i] << (i + 1 == ans.size() ? '\n' : ' ');
        }
    }

    return 0;
}

File: L/submissions/tle/tle.py
#!/usr/bin/python3


def solve(n):
    ans = []

    x = 1
    while x * x * x <= n:
        if n % (x * x + x) == 0:
            ans.append(x)
        x += 1

    if n == 1:
        return ans

    k = 1
    while True:
        m = n // k
        x = int((1 + 4 * m)**0.5) - 1 // 2
        if x**3 < n:
            break
        if n % k == 0 and x * (x + 1) == m:
            ans.append(x)
        k += 1

    return ans


def main():
    t = int(input())
    for cs in range(t):
        n = int(input())
        ans = solve(n)
        ans.sort()
        if len(ans) == 0:
            print(-1)
        else:
            print(*ans)


main()
